### PR TITLE
mjpege: change frame type to MFX_FRAMETYPE_I from MFX_FRAMETYPE_UNKNOWN

### PR DESCRIPTION
**Pull request by Alexandr-Konovalov**
Fixes: #970

### CODE REVIEW

File: _studio/mfx_lib/encode/mjpeg/src/mfx_mjpeg_encode.cpp [hunk 1]
```diff
@@ -659,10 +659,11 @@ mfxStatus MFXVideoENCODEMJPEG::MJPEGENCODECompleteProc(void *pState, void *pPara
     }
 
     pTask->bs->DataLength += (mfxU32)(pDataOut.GetDataSize());
-    
+    pTask->bs->FrameType = MFX_FRAMETYPE_I;
```
Comment (dvrogozh) [inline _studio/mfx_lib/encode/mjpeg/src/mfx_mjpeg_encode.cpp]: I always thought that such frame encoding characteristics as gop structure should be assigned within EncodeFrameAsync and only data is filled once SyncOperation is completed. From this perspective this assignment looks incorrect - the one which you did in EncodeFrameCheck should be enough.

File: _studio/mfx_lib/encode_hw/mjpeg/src/mfx_mjpeg_encode_hw.cpp [hunk 2]
```diff
@@ -972,6 +972,7 @@ mfxStatus MFXVideoENCODEMJPEG_HW::EncodeFrameCheck(
 
     bs->TimeStamp = surface->Data.TimeStamp;
     bs->DecodeTimeStamp = surface->Data.TimeStamp;
+    bs->FrameType = MFX_FRAMETYPE_I;
```
Comment (dvrogozh) [inline _studio/mfx_lib/encode_hw/mjpeg/src/mfx_mjpeg_encode_hw.cpp]: What about SW path? And are there a shared SW/HW code where we can do that?

Comment (Alexandr-Konovalov) [reply _studio/mfx_lib/encode_hw/mjpeg/src/mfx_mjpeg_encode_hw.cpp]: SW path is covered by change in mfx_mjpeg_encode.cpp. I found no right place to put `MFX_FRAMETYPE_I` assignment that is called from `EncodeFrameAsync` and shared by HW and SW pathes.

File: _studio/mfx_lib/encode/mjpeg/src/mfx_mjpeg_encode.cpp (hunk 1 above)
Comment (Alexandr-Konovalov) [reply _studio/mfx_lib/encode/mjpeg/src/mfx_mjpeg_encode.cpp]: Sure, moved to `MFXVideoENCODEMJPEG::EncodeFrameCheck()`.

**Pull request merged**